src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -57,6 +58,13 @@
     
     abstract public SamReader open(final File file);
 
+    public SamReader open(final Path path) {
+        final SamInputResource r = SamInputResource.of(path);
+        final Path indexMaybe = SamFiles.findIndex(path);
+        if (indexMaybe != null) r.index(indexMaybe);
+        return open(r);
+    }
```

Comment:
That should work fine as the implementation of `open` will use the reference sequence set on the factory (if any).